rest/route/host_agent.go

```diff
@@ -375,6 +375,11 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 
 		// validate that the task can be run, if not fetch the next one in the queue.
 		if !nextTask.IsHostDispatchable() {
+			grip.Debug(message.WrapError(err, message.Fields{
+				"investigation": "DEVPROD-12086",
+				"message":       "task was not dispatchable",
+				"task":          nextTask.Id,
```

Comment:
nit: Might be useful to add other info like the variant/project if you want to see if the problem is particularly common in specific projects.

Same with below.

Reply:
updated